src/Form.php

```diff
@@ -0,0 +1,373 @@
+    }
+
+    /**
+     * Populates the model with input data.
+     *
+     * which, with `load()` can be written as:
+     *
+     * ```php
+     * $body = $request->getParsedBody();
+     * $method = $request->getMethod();
+     *
+     * if ($method === Method::POST && $loginForm->load($body)) {
+     *     // handle success
+     * }
+     * ```
+     *
+     * `load()` gets the `'FormName'` from the model's [[formName()]] method (which you may override), unless the
+     * `$formName` parameter is given. If the form name is empty, `load()` populates the model with the whole of
+     * `$data`,
+     * instead of `$data['FormName']`.
+     *
+     * Note, that the data being populated is subject to the safety check by [[setAttributes()]].
+     *
+     * @param array $data the data array to load, typically `$_POST` or `$_GET`.
+     *
+     * @return bool whether `load()` found the expected form in `$data`.
+     */
+    public function load(array $data): bool
+    {
+        $result = false;
+
+        $scope = $this->formName();
+
+        if ($scope === null && !empty($data)) {
+            $this->setAttributes($data);
+            $result = true;
+        } elseif (isset($data[$scope])) {
+            $this->setAttributes($data[$scope]);
+            $result = true;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Sets the attribute values in a massive way.
+     *
+     * @param array $values attribute values (name => value) to be assigned to the model.
+
+     * {@see safeAttributes()}
+     * {@see attributes()}
+     */
+    public function setAttributes(array $values): void
+    {
+        foreach ($values as $name => $value) {
+            if (isset($this->attributes[$name])) {
+                switch ($this->attributes[$name]) {
+                    case 'bool':
+                        $this->$name((bool) $value);
+                        break;
+                    case 'int':
+                        $this->$name((int) $value);
+                        break;
+                    default:
+                        $this->$name($value);
+                        break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the list of attribute names.
+     *
+     * By default, this method returns all private non-static properties of the class.
+     *
+     * @throws \ReflectionException
+     *
+     * @return array list of attribute names.
+     */
+    private function attributes(): array
+    {
+        $class = new \ReflectionClass($this);
+
+        $type = null;
+
+        foreach ($class->getProperties(\ReflectionProperty::IS_PRIVATE) as $property) {
+            if (!$property->isStatic()) {
+                $type = (new \ReflectionProperty($property->class, $property->name))->getType();
+
+                if ($type !== null) {
+                    $this->attributes[$property->getName()] = $type->getName();
```

Comment:
What about to store data in  the following structure:
```php
attributes: [
 {"type": "bool", "value": true, "modifier": "private", "setter": "setMyBestValue", etc...},
 {"type": "object", "value": *object*, "modifier": "public", "setter": null, etc...},
]
```

Reply:
Why?